website/docs/cloud-docs/api-docs/workspaces.mdx

```diff
@@ -1691,7 +1691,7 @@ $ curl \
 
 ## Force Unlock a workspace
 
-This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace.
+This endpoint force unlocks a workspace. Only users with admin access are authorized to force unlock a workspace. Force unlocking the workspace will attempt to find the latest finalized state version to set as the current if there are any provisional state versions available. It is possible that the latest finalized state version is not the most recently created state version.
```

Comment:
does the second sentence mean that it will look for a provisional state snapshot and use that instead of a full state file? if so, will suggest a slight rephrase so it's clear.

also, what are the implications of this for future runs?

Reply:
> does the second sentence mean that it will look for a provisional state snapshot and use that instead of a full state file? if so, will suggest a slight rephrase so it's clear.

State versions are the same as before but now that we've decoupled creating the state version record (i.e the metadata about the statefile) and uploading the actual state file. A state version that is `finalized` has both the record created and state file contents uploaded. A version where the record is created but the contents have not been uploaded is a `pending` state version. The attribute `status` tracks this. 

So during a regular unlock, TFC will ensure that the most recently created state version is finalized before setting it as the current state version for a workspace. If not, it will prevent an unlock. However something can go awry with the statefile upload which leaves the user no choice but to force unlock. In these cases, we will still attempt to find the most recent finalized version, this time without a hard stop at the latest state version. 

> also, what are the implications of this for future runs?
None. All this logic occurs before the next pending run is started. As long as we have a current state version set after an unlock, we're good.